### PR TITLE
fix: grid2: improve the onStart callback and the cancel loop

### PR DESCRIPTION
**Pull request by c9s**

### CODE REVIEW

File: pkg/strategy/grid2/strategy.go [hunk 1]
```diff
@@ -875,48 +875,48 @@ func (s *Strategy) cancelAll(ctx context.Context) error {
...
-			s.logger.Infof("found OrderGroupID (%d), using group ID for canceling orders...", s.OrderGroupID)
+		for {
+			s.logger.Infof("checking %s open orders...", s.Symbol)
 
-			op := func() error {
-				_, cancelErr := service.CancelOrdersByGroupID(ctx, s.OrderGroupID)
-				return cancelErr
-			}
-			err := backoff.Retry(op, backoff.WithMaxRetries(backoff.NewExponentialBackOff(), 101))
+			openOrders, err := session.Exchange.QueryOpenOrders(ctx, s.Symbol)
```
Comment (gx578007) [inline pkg/strategy/grid2/strategy.go:881]: Is this `ctx` constrained with a deadline?

Comment (c9s) [reply pkg/strategy/grid2/strategy.go:881]: not confirmed yet. maybe different when calling CleanUp()

File: pkg/strategy/grid2/strategy.go [hunk 2]
```diff
@@ -875,48 +875,48 @@ func (s *Strategy) cancelAll(ctx context.Context) error {
...
 			if err != nil {
-				werr = multierr.Append(werr, err)
-			}
-		} else {
-			s.logger.Infof("canceling all orders...")
-			op := func() error {
-				_, cancelErr := service.CancelAllOrders(ctx)
-				return cancelErr
+				return err
```
Comment (gx578007) [inline pkg/strategy/grid2/strategy.go:883]: Could it fallback to cancel all orders anyway if having any error? Querying open orders is only for skipping when there is no open orders.

Comment (c9s) [reply pkg/strategy/grid2/strategy.go:883]: it's exp backoff, so if we have error, it will always retry

**Review by gx578007 — APPROVED** [review]

**Pull request merged**